### PR TITLE
fix for infinite recursion on Windows, issue #170

### PR DESCRIPTION
**Pull request by chorny**
So it would be lowercase and match

### CODE REVIEW

File: lib/JSON/Validator.pm [hunk 1]
```diff
@@ -524,7 +524,9 @@ sub _location_to_abs {
   # definitely relative now
   if ($base->isa('Mojo::File')) {
     return $base if !length $location;
-    return $base->sibling(split '/', $location)->realpath;
+    my $tpath = $base->sibling(split '/', $location)->realpath;
```
Comment (jhthorsen) [inline lib/JSON/Validator.pm]: What is "t" in "tpath"?

File: lib/JSON/Validator.pm [hunk 2]
```diff
@@ -524,7 +524,9 @@ sub _location_to_abs {
...
+    #lc fixes https://github.com/mojolicious/json-validator/issues/170
```
Comment (jhthorsen) [inline lib/JSON/Validator.pm]: This comment is not needed.

**Comment by karenetheridge** [comment]
So, it appears (from inferring from the patch provided) that infinite recursion is occuring because two paths are being compared case-sensitively on a case-insensitive filesystem, and they should match but they do not. Where in the tests is that happening? That is, where are two paths provided that are identical-except-for-case?

This also provides a hint for writing a demonstration test -- one could perhaps hack CASE_TOLERANT so it is the same value across all platforms (for this one test), and demonstrate how behaviour is different with and without the `lc`.

I'm also curious where the path comparison is taking place, such that lower-casing is necessary -- because `lc` is not necessarily the right thing to do here, given various locales' handling of case.  When comparing strings, foldcase (`fc`) is preferred.

**Comment by jhthorsen** [comment]
I'm going to close this, since it seems like @charsbar is not interested in this anymore.

**Comment by charsbar** [comment]
@jhthorsen , well maybe it was @chorny not me, but anyway, I sent a modified version of this PR #180

**Comment by jhthorsen** [comment]
That was a bad autocompleted from me. Sorry about that 😢

**Pull request closed without merging**